plugin/lighthouse/setup_test.go

```diff
@@ -22,3 +23,12 @@ var _ = Describe("[setup] Test basic bring up", func() {
 	})
 
 })
+
+func testStart() {
+	BuildConfigFromFlags = func(masterUrl, kubeconfigPath string) (*rest.Config, error) {
+		if kubeconfigPath == "" {
+			return &rest.Config{}, nil
+		}
+		return nil, nil
+	}
+}
```

Comment:
I see now that this is done to override `clientcmd.BuildConfigFromFlags` to avoid errors. However this should really be part of test setup (ie `BeforeEach`) rather than nested in a separate `When` block (which looks a bit odd). I made modifications and added more tests/assertions in this [commit](https://github.com/submariner-io/lighthouse/pull/22/commits/8aa36e88c99f3e6cf61712a7d3518885f9d35a64).